views/default/js/mentions/autocomplete.js

```diff
@@ -87,23 +95,42 @@ define(function(require) {
 		});
 	};
 
-	var autocomplete = function (content, position) {
-		beforeMention = content.substring(0, position);
-		afterMention = content.substring(position);
-		parts = beforeMention.split(' ');
-		current = parts[parts.length - 1];
-
-		precurrent = false;
-		if (parts.length > 1) {
-			precurrent = parts[parts.length - 2];
-			if (!current.match(/@/)) {
-				if (precurrent.match(/@/)) {
-					current = precurrent + ' ' + current;
+	var autocomplete = function (content, position_or_data) {
+		if (mentionsEditor == 'ckeditor') {
+			edata = position_or_data;
+			current = "";
+			if (content.length != 0) {
+				//Split to get last item
+				var words = content.split(/ +/);
+				var html_words = edata.split(/ +|\r\n|\n/);
+				var last_word = words[words.length -1];
+				var last_html_word = html_words[html_words.length -2];
+				if (last_word.match(/^@\w{3,}/)&& (last_html_word == last_word+'</p>' || last_html_word == '<p>'+last_word+'</p>')) {
+					var search_word = last_word.replace('@','');
+					if (! search_word.match(/\W/)) {
+						current = last_word;
+					};
+				};
+			};
+		} else {
+			position = position_or_data;
+			beforeMention = content.substring(0, position);
+			afterMention = content.substring(position);
+			parts = beforeMention.split(' ');
+			current = parts[parts.length - 1];
+
+			precurrent = false;
+			if (parts.length > 1) {
+				precurrent = parts[parts.length - 2];
+				if (!current.match(/@/)) {
+					if (precurrent.match(/@/)) {
+						current = precurrent + ' ' + current;
+					}
 				}
 			}
 		}
 
-		if (current.match(/@/) && current.length > 1) {
+		if (current.match(/@/) && current.length > 4) {
```

Comment:
I presume that 4 is the default value for min username length, as seen in `validate_username()`:

```php
        if (!isset($CONFIG->minusername)) {
		$CONFIG->minusername = 4;
	}
```

Ideally we would inject whatever that config value is, but it might be a hassle. Perhaps adding a note as to why this is 4, would suffice.